### ISSUE TITLE
Display placeholder for images with no existing thumbnail

### ISSUE THREAD

**Issue by mmccarn** [nextcloud/gallery#229]
Suggestion:

Modify the thumbnail generation code to return a placeholder immediately and queue a background job to create the actual thumbnail.

Advantages:
* No delay when displaying a new collection
* no need for command line thumbnail generation (which according to the [placeholder wiki page](https://github.com/nextcloud/gallery/wiki/Create-thumbnails) will make the affected folder and its sub-folders inaccessible during operation)
* the affected user is already logged in, so there shouldn't be any issues with authentication or encryption.

Difficulties:
* I have no idea if Nextcloud includes any facilities for creating and managing a background queue as proposed.
* background thumbnail generation would have to be done "politely" to avoid impacting server performance.
* The user's screen should be refreshed periodically, or if a given number of thumbnails in the currently selected folder are completed (although I suspect those of us who are running into thumbnail generation problems would also be happy to hit 'refresh' from time to time...)
* It would be convenient if there was an indicator showing that background thumbnail generation was in progress for a the current folder

**Comment by oparoz** [comment]
We can't really do that because of the way we build the photowall.
We need to build it row by row.
We send several requests for thumbnails and rows are completed as soon as all of them have landed.

**Comment by mmccarn** [comment]
Thanks.

With my proposal the user would quickly get the entire wall filled with placeholders and the job of creating the missing thumbnails would be delegated to a background task on the server. 

In my case, the row-by-row thumbnail generation has been pretty annoying.  I uploaded a collection of photos with about 90,000 jpgs.  I've spent lots of time  loading each folder one at a time, scrolling to the bottom, wait, scroll, wait, scroll... but I'm still finding folders without thumbnails from time to time, which I think is slowing down the time it takes to load the root folder in Gallery.

(It still takes a little over 3 minutes to load the root photo folder in gallery)

Of course,  I understand if the pieces don't fit together in a way that makes it easy to solve _my_ problem...

**Comment by mmccarn** [comment]
Update:

I reduced my gallery load time from 3+ minutes (190 seconds) to 6 seconds.

I found a few folders that did not yet have any thumbnails generated.

However, the biggest difference came from adding a ".nomedia" file to folders with lots of files in them that didn't contain anything that Gallery needed to process. 

In my case, I have two iPhoto Libraries on my nextcloud server, each of which contains several folders that don't contain any useful image information.

After doing this in each iPhoto library folder, my gallery performance is great:

```
touch Attachments/.nomedia
touch Thumbnails/.nomedia
touch Previews/.nomedia
touch Auto\ Import/.nomedia
touch iLifeShared/.nomedia
touch Database/.nomedia
chown www-data:www-data */.nomedia
```
The "Database" folder seemed to make the most difference - in one Library I had 21K folders containing 70K files; in the other I had 7K folders containing 23K files

Perhaps a config or occ option for 'Gallery' to add ".nomedia" to any folder tree that:
* contains no files that will be handled by Gallery
* (optionally) contains only Gallery-eligible files that are at or below the Gallery thumbnail resolution
